packages/dx-react-scheduler-material-ui/src/templates/date-navigator/calendar/root.jsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import moment from 'moment';
+import { Table } from './table';
+
+export class Root extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    const { currentDate, onNavigate } = this.props;
+    this.state = {
+      currentDate,
+    };
+    this.onNavigate = ({ back }) => {
```

Comment:
Why do we inline methods in the constructor? It may lead to implicit closures on initial state or props variables. Use normal syntax where applicable.